api/organisations/models.py

```diff
@@ -237,17 +241,59 @@ def update_mailer_lite_subscribers(self):
             mailer_lite = MailerLite()
             mailer_lite.update_organisation_users(self.organisation.id)
 
-    def cancel(self, cancellation_date=timezone.now(), update_chargebee=True):
-        self.cancellation_date = cancellation_date
+    def save_as_free_subscription(self):
+        """
+        Wipes a subscription to a normal free plan.
+
+        The only normal field that is retained is the notes field.
+        """
+        self.subscription_id = None
+        self.subscription_date = None
+        self.plan = FREE_PLAN_ID
+        self.max_seats = MAX_SEATS_IN_FREE_PLAN
+        self.max_api_calls = MAX_API_CALLS_IN_FREE_PLAN
+        self.cancellation_date = None
+        self.customer_id = None
         self.billing_status = None
+        self.payment_method = None
+
         self.save()
-        # If the date is in the future, a recurring task takes it.
-        if cancellation_date <= timezone.now():
-            self.organisation.cancel_users()
+
+        if not getattr(self.organisation, "subscription_information_cache", None):
+            return
+
+        self.organisation.subscription_information_cache.delete()
+
+    def prepare_for_cancel(
+        self, cancellation_date=timezone.now(), update_chargebee=True
+    ) -> None:
+        """
+        This method get's a subscription ready for cancelation.
+
+        If cancellation_date is in the future some aspects are
+        reserved for a task after the date has passed.
+        """
+        # Avoid circular import.
+        from users.models import FFAdminUser
 
         if self.payment_method == CHARGEBEE and update_chargebee:
             cancel_chargebee_subscription(self.subscription_id)
 
+        FFAdminUser.send_alert_to_admin_users(
```

Comment:
nit: can we move this to a private method or maybe a task that we can delay and send emails asynchronously ... what do you think?

Reply:
I think a small task would be a good idea. Let me implement that.